avocado/core/task/runtime.py

```diff
@@ -46,3 +55,132 @@ def __eq__(self, other):
         if isinstance(other, RuntimeTask):
             return hash(self) == hash(other)
         return False
+
+
+class RuntimeGraph:
+
+    def __init__(self, test_suite, job_id, status_server_uri):
+        self.test_suite = test_suite
+        self.job_id = job_id
+        self.status_server_uri = status_server_uri
+
+    def _get_requirements_runtime_tasks(self, runnable, prefix):
```

Comment:
I'll leave it to the author (@richtja) to give his view about the overall suggestion. 

But, I have a question to @beraldoleal.  I see the reasoning behind gathering `RuntimeTask` related code in the same class, but, depending on how you interpret The Single Responsibility Principle, it may actually point you towards the opposite direction (having different classes for different responsibilities).  See, a list of a `RuntimeTask`s is not *a* `RuntimeTask`, only made "similar" because Python has such easy to use lists, and the responsibility if quite different.

For instance, on Wikipedia, the [example](https://en.wikipedia.org/wiki/Single-responsibility_principle#Example) given is:

_"...consider a module that compiles and prints a report. Imagine such a module can be changed for two reasons. First, the content of the report could change. Second, the format of the report could change. These two things change for different causes. The single-responsibility principle says that these two aspects of the problem are really two separate responsibilities, and should, therefore, be in separate classes or modules. It would be a bad design to couple two things that change for different reasons at different times."_


The author of that principle had a bit of trouble explaining what he really meant, so I may be confused too. :)